test/unit/org/apache/cassandra/utils/SchemaIsolator.java

```diff
@@ -0,0 +1,116 @@
+package org.apache.cassandra.utils;
+
+import java.io.ByteArrayInputStream;
+import java.io.ObjectInputStream;
+import java.lang.reflect.Constructor;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
+import net.bytebuddy.implementation.MethodDelegation;
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.exceptions.UnknownTableException;
+import org.apache.cassandra.schema.Schema;
+import org.apache.cassandra.schema.TableId;
+import org.apache.cassandra.schema.TableMetadata;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+/**
+ * Mocks out some of the methods on {@link Schema} to be easier for tests to manipulate.
+ *
+ * @see #addTable(TableMetadata)
+ * @see #removeTable(TableMetadata)
+ */
+public final class SchemaIsolator implements Isolator
+{
+    private static final ConcurrentMap<TableId, TableMetadata> TABLES = new ConcurrentHashMap<>();
+
+    private SchemaIsolator()
+    {
+
+    }
+
+    public static Isolator create()
+    {
+        IsolatorClassLoader cl = new IsolatorClassLoader();
+
+        // redefine Schema
+        new ByteBuddy().redefine(Schema.class)
+                       .method(named("getExistingTableMetadata"))
+                       .intercept(MethodDelegation.to(SchemaIsolator.class))
+                       .make()
+                       .load(cl, ClassLoadingStrategy.Default.INJECTION);
+
+        // load in class loader
+        try
+        {
+            Class<?> klass = cl.loadClass("org.apache.cassandra.utils.SchemaIsolator");
+            Constructor<?> c = klass.getDeclaredConstructor(new Class[0]);
+            c.setAccessible(true);
+            return (Isolator) c.newInstance(new Object[0]);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void checkState()
+    {
+        if (!Lazy.isLoaded())
+            throw new IllegalStateException("Unable to redefine Schema");
+    }
+
+    private <T> T load(byte[] bytes) throws Exception
+    {
+        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes);
+             ObjectInputStream ois = new ObjectInputStream(bais))
+        {
+            return (T) ois.readObject();
+        }
+    }
+
+    public void transferAndRunExecute(byte[] bytes) throws Exception
+    {
+        checkState();
+        Execute execute = load(bytes);
+        execute.execute();
+    }
+
+    /**
+     * Used by ByteBuddy when ever {@link Schema#getExistingTableMetadata(TableId)} is called
+     */
+    @UsedByByteBuddy("org.apache.cassandra.schema.Schema")
```

Comment:
Do we actually do any meta-programming w/ `UsedByByteBuddy`, or is it primarily a documentation mechanism at this point? I'd be okay removing it if it's the latter.

Reply:
it was just for docs.  The code is marked dead code in IntelliJ so added this so readers of the code knew they couldn't remove.

I will remove for now.